File: plugin/src/main/java/io/papermc/hangarpublishplugin/HangarAuthorizationToken.java
package io.papermc.hangarpublishplugin;

import java.util.concurrent.TimeUnit;

public final class HangarAuthorizationToken {
    private final String jwt;
    private final long expiresAt;

    public HangarAuthorizationToken(final String jwt, final long expiresAt) {
        this.jwt = jwt;
        this.expiresAt = expiresAt;
    }

    public boolean shouldRenew() {
        // Request a new one a little before the active one expires
        return System.currentTimeMillis() < this.expiresAt + TimeUnit.SECONDS.toMillis(3);
    }

    public String jwt() {
        return jwt;
    }

    public long expiresAt() {
        return expiresAt;
    }
}

File: plugin/src/main/kotlin/io/papermc/hangarpublishplugin/internal/HangarAuthorizationToken.kt
package io.papermc.hangarpublishplugin.internal

import java.util.concurrent.TimeUnit

data class HangarAuthorizationToken(val jwt: String, val expiresAt: Long) {
    companion object {
        fun create(jwt: String, expiresIn: Long): HangarAuthorizationToken =
            HangarAuthorizationToken(jwt, System.currentTimeMillis() + expiresIn)
    }

    fun shouldRenew(): Boolean {
        // Request a new one a little before the active one expires
        return System.currentTimeMillis() < expiresAt + TimeUnit.SECONDS.toMillis(3)
    }
}
